src/libcore/tests/ptr.rs

```diff
@@ -62,6 +63,28 @@ fn test_is_null() {
 
     let mq = unsafe { mp.offset(1) };
     assert!(!mq.is_null());
+
+    // Pointers to unsized types
+    let s: &mut [u8] = &mut [1, 2, 3];
+    let cs: *const [u8] = s;
+    assert!(!cs.is_null());
+
+    let ms: *mut [u8] = s;
+    assert!(!ms.is_null());
+
+    let cz: *const [u8] = &[];
+    assert!(!cz.is_null());
+
+    let mz: *mut [u8] = &mut [];
+    assert!(!mz.is_null());
+
+    unsafe {
+        let ncs: *const [u8] = slice::from_raw_parts(null(), 0);
```

Comment:
This is UB (null in a `&[u8]`) so unfortunately can begin to fail when the compiler gets smarter.

Reply:
Is there any better way to get a null unsized pointer?  Or are they impossible to be null in practice?  If the latter, we could just drop those parts of the tests.

Reply:
I suppose I could implement a local `Repr` hack for the tests, similar to `slice::from_raw_parts` but without ever calling it a `&`.

Reply:
Depending on how much you hate `transmute`, you could do `let ncs: *const [u8] = unsafe { mem::transmute((0usize, 0usize)) };`

Reply:
I went ahead and added a `transmute` with `Repr`, which I think is a little safer than a tuple because it can be `#[repr(C)]`.  This is all just for testing anyway, so whatever we do isn't going to affect anyone in the wild.